service/matching/taskReader.go

```diff
@@ -140,6 +140,7 @@ dispatchLoop:
 					tr.tlMgr.logger.Info("Taskqueue manager context is cancelled, shutting down")
 					return err
 				}
+				// todo: if err is != nil and != context.Canceled, we don't handle it here
```

Comment:
@dnr I assume that's because we just want to log these errors and retry?

Reply:
right, we do handle it here: by retrying